stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/StreamFileUtil.kt

```diff
@@ -54,4 +58,54 @@ public object StreamFileUtil {
             null
         }
     }
+
+    /**
+     * Hashes the links of given attachments and then tries to create a new file
+     * under that hash. If the file already exists checks that the full file
+     * has been written and shares it if it has, in other cases downloads the file
+     * and writes it.
+     *
+     * @param context The Android [Context] used for path resolving and [Uri] fetching.
+     * @param attachment the attachment to be downloaded.
+     *
+     * @return The [Uri] that represents the path to the downloaded file.
+     */
+    public suspend fun writeFileToShareableFile(context: Context, attachment: Attachment): Uri? {
+        val result = runCatching {
+            val attachmentName = (attachment.url ?: attachment.assetUrl)?.hashCode()
+            val fileName = attachmentName.toString() + attachment.name
+
+            val file = File(context.cacheDir, fileName)
```

Comment:
When do we clean this cache?
I see a couple of ways of doing it.

### 1. Clean the cache when the gallery activity closes

This way we clean it regularly, and if we miss a lifecycle event it's not a bit deal because the chances are we'll clean it up the next time.

Con is that going back and reopening the gallery will make the file have to be re downloaded.

### 2. Expose cache directories to users and warn users about cleaning them in the docs

The pros are that we are absolved of having to manage cache manually, which can be tricky for an SDK library and that we give the users the option to clean these manually.

The con is obvious, with the fact that they have the option comes the responsibility to do so and the lack of a plug'n'play feeling.

Reply:
We could have a subfolder in the cache directory called `StreamCache` that would be exposed for cleaning.

Reply:
The first option is a really bad experience for the final user.
The second option aport almost nothing to our implementation, because by default we are not exposing any "clearign mechanism"

Reply:
I like @JcMinarro idea, exposing `CacheManager` or `CachePolicy` with the default implementation sounds like a good plan

Reply:
@JcMinarro  @adasiewiczr  

Yup, I think we are all on the same page here.

That was essentially the initial idea, but I expressed it poorly here. Not to expose the path itself but rather cache actions through normal functions such as `saveFile`, `deleteCache`, etc.

As discussed in today's meeting, we have a relatively clear heading.

Thanks for the input guys! <3